### PR TITLE
Update Java dependency management page

### PR DESCRIPTION
**Pull request by rszper**
* Add a section with information about dependency management from the blog post: https://beam.apache.org/blog/managing-beam-dependencies-in-java/
* Reorganize the existing content to make it clearer

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by rszper** [comment]
R: @bvolpato

File: website/www/site/content/en/documentation/sdks/java-dependencies.md [hunk 1]
```diff
@@ -18,29 +18,52 @@ limitations under the License.
...
+When problems occur with dependencies, you might see unexpected behavior in the service,
+including errors such as `NoClassDefFoundError`, `NoSuchMethodError`, `NoSuchFieldError`,
+or `FATAL ERROR in native method`.
+
+This page explains how to view the dependencies that your SDK is using and how to manage your
+dependencies to avoid issues.
+
+## View dependencies
+
+To view your dependencies, either use the
```
Comment (bvolpato) [inline website/www/site/content/en/documentation/sdks/java-dependencies.md:38]: linter: trailing space

File: website/www/site/content/en/documentation/sdks/java-dependencies.md [hunk 2]
```diff
@@ -18,29 +18,52 @@ limitations under the License.
...
+## View dependencies
+
+To view your dependencies, either use the 
+`BeamModulePlugin.groovy` file or retrieve the list by creating a new project
+through Maven and resolving the dependencies.
+
+### Use BeamModulePlugin.groovy to retrieve dependencies
+
+The `BeamModulePlugin.groovy` file in the Beam repository lists compile and runtime
+dependencies for your Beam SDK version.
```
Comment (bvolpato) [inline website/www/site/content/en/documentation/sdks/java-dependencies.md:45]: linter: trailing space

**Review by bvolpato — APPROVED** [review]
LGTM, thanks -- just added 2 comments related to the linter

```
./website/www/site/content/en/documentation/sdks/java-dependencies.md:38:41: W201 Trailing whitespace
./website/www/site/content/en/documentation/sdks/java-dependencies.md:45:39: W201 Trailing whitespace
```

File: website/www/site/content/en/documentation/sdks/java-dependencies.md (hunk 1 above)
Comment (rszper) [reply website/www/site/content/en/documentation/sdks/java-dependencies.md]: Fixed, thanks for catching these.

**Review by bvolpato — APPROVED** [review]
LGTM, thank you!

**Pull request merged**